### ISSUE TITLE
RemoteApp not working on Windows 7 Pro

### ISSUE THREAD

**Issue by aiugrivef** [binarymaster/rdpwrap#13]
Hello,

I can see in your code that you change some variables for RemoteApp, so I tested RemoteApp on Windows 7 Pro x86 (6.1.7601.18637).
However, this is not working at all.

Full Desktop remote sessions are working fine, but RemoteApp not.
Do you have any idea on how this could work ?

Also, do you know how to activate Debug Messages for termsrv.dll ? I wanted to help you, but I cannot see these Debug Messages with WinDbg...

**Comment by v-yadli** [comment]
I'm also having some RemoteApp issues with Windows 10 TP2 build 9926. Before the update I was staying with 9874 and everything runs smoothly. Now RemoteApp refuses to start initially, but once a full rdp session is established, it restores functionality. Maybe, I guess they have brought in yet another jmp instruction for us to patch under certain circumstances.. :-D

**Comment by mildik-dik** [comment]
Enable RDP -> Programs -> "Start the following program on conection"

Regedit:

Windows Registry Editor Version 5.00
[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows NT\Terminal Services]
"fAllowUnlistedRemotePrograms"=dword:00000001
;"fTurnOffSingleAppMode"=dword:00000000
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Terminal Server\TSAppAllowList]
"HonorLegacySettings"=dword:00000001
[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Terminal Server]
"HonorLegacySettings"=dword:00000001

**Comment by sharo1k** [comment]
binarymaster, except all things that you mentioned, you have to patch rdpinit.exe (1.7601.17514) with following:
0000568E: 04 01
0000568F: 40 00
00005691: 80 00
This is return value of checking ...ClientSku-RAILAllowed policy. Assuming that policy is missing.

**Comment by AlexeiScherbakov** [comment]
Windows 10 x64 One Language - RemoteApp is not working (Version 1.6).
RDP working normally

Can anybody provide working solution for RemoteApp without installing Windows Ultimate?

**Comment by neggenbe** [comment]
Great, great library, thanks!
Will RemoteApp be supported officially sometime?

**Comment by binarymaster** [comment]
@neggenbe maybe sometime.

If someone offers a patch for `termsrv`, which overcomes this issue, I'll implement it in the wrapper and for each possible version in INI file.

**Comment by neggenbe** [comment]
ok great, thanks.
Another topic - Is there any discussion about conflicts with Windows licence terms and so on?

**Comment by binarymaster** [comment]
> Another topic - Is there any discussion about conflicts with Windows licence terms and so on?
https://github.com/binarymaster/rdpwrap/issues/26

**Comment by trockenasche** [comment]
@sharo1k I tried to patch the rdpinit.exe with a hex editor, but I don't get it why you have a 2 byte value?
Shouldn't it 1 byte value to change in HEX? Or am I wrong?

**Comment by sharo1k** [comment]
@trockenasche:
these are old byte->new byte. This is return value of function which checks for TerminalServices-.....-RAILAllowed policy. With this patch, the function retruns 0x01 instead of 0x80004004

**Comment by chrwick** [comment]
Could you please give a short summary about the requirements that are required to get remote app working in case of Windows10?
What registry keys are required? What DLLs/EXEs need to be modified?

**Comment by chrwick** [comment]
Just for your info. Today I have tried to use the RemoteApp feature of my new clean Windows 10 Pro 64-Bit installation. I haved used the RemoteApp tool from Kim Knight to configure some programe for remote usage (notepad.exe, cmd.exe). 

I was able to connect to my Windows 10 Pro using the terminal service client that comes with Windows 7 Pro and using the generated RDP file. The was no closing of the connection or similar things. It just works. A single app window shows up on the client. From the CMD.exe window I was able to start other single window apps as well which were not included in the RemoteApp tool list. I had set the host option to allow unlisted apps.

I have used Windows 10 Pro **1607** 64-Bit German ISO from Microsoft. As license key I have used my Windows 7 Pro (SystemBuilder) key which I have chosen to upgrade to Windows 10 Pro during MS free offer. I had entered the license key AFTER the installation was finished and not right at the beginning (don't know if that matters!).

So for me there is was no patching necessary!

Did MS change something with the latest updates?

**Comment by AlexeiScherbakov** [comment]
RemoteApp is working in Pro/Ent. It is not working with Home/One Language
(like RDP itself)

8 Ноя 2016 г. 20:17 пользователь "chrwick" <notifications@github.com>
написал:

**Comment by tostercx** [comment]
@AlexeiScherbakov that is not correct - by default RemoteApp only works on Ent/Ult/Edu versions, not Pro.

**Comment by AlexeiScherbakov** [comment]
Hm, that is iteresting. I don't know about education version...

30 Ноя 2016 г. 2:13 пользователь "tostercx" <notifications@github.com>
написал:

**Comment by kim0** [comment]
Excuse me, it seems most of the comments are focused on non-server Windows OS. My question is, How do I get remote-app functionality on server-2016? 
 * Should I just install rdpwrap+kim's tool ? 
 * Would trying to install a full RDSH setup, with rdpwrap be a good idea?
Thanks folks!

**Comment by AlexeiScherbakov** [comment]
Rdpwrap is only needed on home/one-language versions of windows.
Server systems have this functional by default (2 admin sessions) and with
Remotw Desktop Services(req. terminal server licences). But this server
must be member in domain to enable RemoteApp

6 Дек 2016 г. 15:32 пользователь "Ahmed Kamal" <notifications@github.com>
написал:

**Comment by binarymaster** [comment]
@kim0 try it yourself, maybe it will work out-of-the-box with installed RDP Wrapper.

**Comment by kim0** [comment]
I actually played with it, and it seemed to work. The server is a test domain controller, and I was able to connect 5+ users to it concurrently. I also tested the remote-app software recommended in another bug, which also worked beautifully. So great work folks I guess! I guess the readme should make it clear that it works as well on server OSs.

**Comment by piyushnath** [comment]
My users are not able to connect after 15 concurrent sessions.

**Comment by batminson** [comment]
My users cannot connect after there are 15 simultaneous sessions, how can this problem be solved? my configuration is as follows
Wrapper ver. 1.5.0.0
Service ver. 10.0.18362.657

**Comment by medinaelvis-collab** [comment]
> Excuse me, it seems most of the comments are focused on non-server Windows OS. My question is, How do I get remote-app functionality on server-2016?
> 
> * Should I just install rdpwrap+kim's tool ?
> * Would trying to install a full RDSH setup, with rdpwrap be a good idea?
> 
> Thanks folks!

RDPwrapper won't work under server 2016 you need to run and install RDP and RDS for remote desktop connections, also you would to license the rdp server and cal licenses.
the following website can assist with those licenses
https://thecatontheceiling.github.io/LyssaRDSGen/
